dags/miovision_pull.py

```diff
@@ -0,0 +1,190 @@
+                 "SELECT miovision_api.create_yyyy_volumes_15min_partition('volumes_15min', '{{ macros.ds_format(ds, '%Y-%m-%d', '%Y') }}'::int)",
+                 "SELECT miovision_api.create_yyyy_volumes_15min_partition('volumes_15min_mvt', '{{ macros.ds_format(ds, '%Y-%m-%d', '%Y') }}'::int)"],
+            postgres_conn_id='miovision_api_bot',
+            autocommit=True
+        )
+      
+        create_month_partition = PostgresOperator(
+            task_id='create_month_partition',
+            sql="""SELECT miovision_api.create_mm_nested_volumes_partitions('volumes'::text, '{{ macros.ds_format(ds, '%Y-%m-%d', '%Y') }}'::int, '{{ macros.ds_format(ds, '%Y-%m-%d', '%m') }}'::int)""",
+            postgres_conn_id='miovision_api_bot',
+            autocommit=True
+        )
+
+        check_jan_1st.override(task_id="check_annual_partition")() >> create_annual_partition >> (
+            check_1st_of_month.override(task_id="check_month_partition")() >> create_month_partition
+        )
+
+    @task(trigger_rule='none_failed', retries = 1)
+    def pull_miovision(ds = None, **context):
+        if context["params"]["intersection"] == [0]:
+            INTERSECTION = ()
+        else:
+            INTERSECTION = tuple(context["params"]["intersection"])
+        run_api(start_date=ds,
+                end_date=ds_add(ds, 1),
+                path=API_CONFIG_PATH,
+                intersection=INTERSECTION,
+                pull=True,
+                dupes=True)
+
+    @task_group
+    def miovision_agg():
+        @task
+        def find_gaps_task(ds = None):
+            mio_postgres = PostgresHook("miovision_api_bot")  
+            with mio_postgres.get_conn() as conn:
+                find_gaps(conn, ds, ds_add(ds, 1))
+
+        @task
+        def aggregate_15_min_mvt_task(ds = None, **context):
+            mio_postgres = PostgresHook("miovision_api_bot")
+            if context["params"]["intersection"] != 0:
+                intersections = get_intersection_info(conn, intersection=context["params"]["intersection"])
+                with mio_postgres.get_conn() as conn:
+                    aggregate_15_min_mvt(conn, ds, ds_add(ds, 1), True, intersections)
+            else:
+                with mio_postgres.get_conn() as conn:
+                    aggregate_15_min_mvt(conn, ds, ds_add(ds, 1))
+
+        @task
+        def aggregate_15_min_task(ds = None):
+            mio_postgres = PostgresHook("miovision_api_bot")  
+            with mio_postgres.get_conn() as conn:
+                aggregate_15_min(conn, ds, ds_add(ds, 1))
+
+        @task
+        def aggregate_volumes_daily_task(ds = None):
+            mio_postgres = PostgresHook("miovision_api_bot")  
+            with mio_postgres.get_conn() as conn:
+                aggregate_volumes_daily(conn, ds, ds_add(ds, 1))
+
+        @task
+        def get_report_dates_task(ds = None):
```

Comment:
Is there a reason why get_report_dates is here but the api_log one is naut?

Reply:
yea I feel like we don't (?) really use these three tables, https://github.com/CityofToronto/bdit_data-sources/tree/master/volumes/miovision#other-important-tables

Reply:
I feel like `report_dates` could be useful for reporting.

`api_log` is more of a record of each data pull... what were the start and end parameters and when that data was inserted into the database. I initially was thinking it's not that helpful, but could be (rarely) handy like the issue we had recently of data missing for some intersections but it had been so long that Airflow logs didn't exist for those dates anymore.